### PR TITLE
Timeseries: Time regions migration

### LINKED ISSUE #66997: Timeseries: Time regions migration

### PR DESCRIPTION
**Pull request by adela-almasan**
Updated migrations to include time regions. Time regions PR needs to be merged first.

Check:  http://localhost:3000/d/XMjIZPmik/panel-tests-graph-time-regions?orgId=1&from=now-30d&to=now&__feature.autoMigrateOldPanels=true

Fixes #66997

image

TO DO:
- [x] auto refresh dashboard?

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by nmarrs — APPROVED** [review]
code changes LGTM - will want to test again after blocking PR is merged

File: public/app/plugins/panel/graph/tab_display.html
Comment (leeoniya) [inline public/app/plugins/panel/graph/tab_display.html:12]: this is outdated. we support negative transforms in TimeSeries field overrides.

Comment (adela-almasan) [reply public/app/plugins/panel/graph/tab_display.html:12]: oh, didn't know that. Is there anything left that's not supported?

File: public/app/plugins/panel/timeseries/migrations.ts
Comment (ryantxu) [inline public/app/plugins/panel/timeseries/migrations.ts:68]:
```suggestion
```

old?

File: public/app/plugins/panel/timeseries/migrations.ts
Comment (ryantxu) [inline public/app/plugins/panel/timeseries/migrations.ts:410]:
```suggestion
        enable: true,
        hidden: true,  // don't show the toggle at the top of the dashboard
```

Comment (adela-almasan) [reply public/app/plugins/panel/timeseries/migrations.ts:410]: hm, isn't that a bit restrictive? in old graph you can set the fill to false to be able to hide them 🤔

File: public/app/plugins/panel/timeseries/migrations.ts
Comment (ryantxu) [inline public/app/plugins/panel/timeseries/migrations.ts:71]: @dprokop @torkelo -- any suggestions on how to trigger the annotations to rerun.  This was my hack, but not working consistently

Comment (torkelo) [reply public/app/plugins/panel/timeseries/migrations.ts:71]: @ryantxu the problem here is that this will be triggered multiple times if you have many panels with time regions on dashboard. So maybe need some kind of debounce function for this. But ideally it should only retrigger annotations. Think it could be better if this was something like (But still need to figure out the debounce), the other problem I guess is that when loading a dashboard this is kind of too late. The annotation queries might already be running. Is there a way to do these migrations in the DahboardMigrator instead so they happen before dashboard is initialized and annotation queries are issued? 

```ts
getDashboardQueryRunner().run({dashboard, range: getTimeSrv().getTimeRange()}
```

**Review by ryantxu — APPROVED** [review]

File: public/app/core/components/Page/Page.tsx
Comment (torkelo) [inline public/app/core/components/Page/Page.tsx:54]: @adela-almasan can you open a PR that removes this, this is the main Page component used by ALL pages in Grafana it should not have a test id of dashboard.wrapper

Comment (nmarrs) [reply public/app/core/components/Page/Page.tsx:54]: https://github.com/grafana/grafana/pull/67402

**Pull request merged**
